### DISCUSSION TITLE
Conduct Instances

### DISCUSSION THREAD

**Discussion by ghost** [z1pti3/jimi#70] [Ideas]
jimiFlows are currently automated segments which do not preserve state or data passed through once a given jimiFlow has completed. I was wondering if adding the ability to create an instance would enable a nice UI interface for tracing back automated actions and incident response? 

I imagine if would save the data in and out of each object so you could interactivity use the conduct to view the response and data collected by the jimiFlow - think it may help with debugging as well as responding to incidents with the platform.

**Comment by b1scuit-thi3f** [comment]
I think this would be a really nice feature. From a debugging perspective it would be highly useful as when things go wrong a visual indicator of where logic might have failed would be quicker to identify than via current methods (search through SIEM/log output).

Would these instances run their triggers through the general worker threads or would they use their own resource pools?

**Comment by ghost** [comment]
When we 1st implemented codify it ran outside the main engine but ended up causing issues with integrations i.e forEach - I feel that is may be best to add the options to the core engine to support the interaction and feedback so that we dont end up with incompatibilities
